test/testrunner.py

```diff
@@ -259,6 +260,15 @@ def main():
     benchmark_mode = "benchmark" in test_type or is_benchmark_dev_context()
     specific_test_type = re.sub("benchmark-", "", test_type) if "benchmark" in test_type else test_type
     build_context = get_build_context()
+    # PR test executions can be distinguished by commit id. 
+    # But other pipelines triggered with the same commit id. 
+    # Adding codebuild_initiator to a pipeline name for distinguishing different build pipelines 
+    if not "PR" == build_context:
+        codebuild_initiator = get_codebuild_initiator()
+        if codebuild_initiator:
+            build_context += f"_{codebuild_initiator}"
```

Comment:
I don't think we should be changing `build_context` like this - as it's also used for eks cluster creation, plus adding codebuild initiator isn't really a part of the build context as we've defined it in different parts of the code. 

Can we add a new param to pytest_cache_params, or define a different variable for this?